index.js

```diff
@@ -780,22 +734,129 @@
+  //.     type; and
+  //.   - `$.Function([a, b, a])` represents the `(a, b) -> a` type.
+  function Function_(types) {
+    function format(outer, inner) {
+      var xs = types.map(function(t, idx) {
+        return unless(t.type === RECORD || isEmpty(t.keys),
+                      stripOutermostParens,
+                      inner('$' + String(idx + 1))(String(t)));
+      });
+      var parenthesize = wrap(outer('('))(outer(')'));
+      return parenthesize(unless(types.length === 2,
+                                 parenthesize,
+                                 init(xs).join(outer(', '))) +
+                          outer(' -> ') +
+                          last(xs));
+    }
+
+    var $keys = [];
+    var $types = {};
+    types.forEach(function(t, idx) {
+      var k = '$' + String(idx + 1);
+      $keys.push(k);
+      $types[k] = {extractor: K([]), type: t};
+    });
+
+    return new _Type(FUNCTION,
+                     '',
+                     '',
+                     format,
+                     AnyFunction,
+                     K(true),
+                     $keys,
+                     $types);
+  }
+
+  //# NonEmpty :: Type -> Type
+  //.
+  //. Constructor for non-empty types. `$.NonEmpty($.String)`, for example, is
+  //. the type comprising every [`String`][] value except `''`.
+  //.
+  //. The given type must satisfy the [Monoid][] and [Setoid][] specifications.
+  var NonEmpty = UnaryType(
+    'sanctuary-def/NonEmpty',
+    functionUrl('NonEmpty'),
+    Any,
+    function(x) {
+      return Z.Monoid.test(x) &&
+             Z.Setoid.test(x) &&
+             !Z.equals(x, Z.empty(x.constructor));
+    },
+    function(monoid) { return [monoid]; }
   );
 
-  //# ValidNumber :: Type
+  //# Nullable :: Type -> Type
   //.
-  //. Type comprising every [`Number`][] value except `NaN`.
-  var ValidNumber = NullaryTypeWithUrl(
-    'sanctuary-def/ValidNumber',
-    function(x) { return Number_._test(x) && !isNaN(x); }
+  //. Constructor for types which include `null` as a member.
+  var Nullable = UnaryTypeWithUrl(
+    'sanctuary-def/Nullable',
+    Any,
+    K(true),
+    function(nullable) {
+      // eslint-disable-next-line eqeqeq
+      return nullable === null ? [] : [nullable];
+    }
+  );
+
+  //# Pair :: Type -> Type -> Type
+  //.
+  //. Constructor for tuple types of length 2. Arrays are said to represent
+  //. tuples. `['foo', 42]` is a member of `Pair String Number`.
+  var Pair = BinaryTypeWithUrl(
+    'sanctuary-def/Pair',
+    Array_(Any),
+    function(x) { return x.length === 2; },
+    function(pair) { return [pair[0]]; },
+    function(pair) { return [pair[1]]; }
+  );
+
+  //# Strict :: Type -> Type
+  //.
+  //. Constructor for strict record types. For example:
+  //. `$.Strict($.Record($.Any, {name: $.String}))`, is the type comprising
+  //. every object with a `name :: String` property, without any other
+  //. properties.
```

Comment:
Do you like this variant?

_Constructor for strict record types._

_`$.Strict($.Record($.Any, {name: $.String}))`, for example, is the type comprising every object with exactly one field, `name`, of type `String`._